File: peer-web/src/peer-config/peer-auth.ts
export interface PeerAuth {
    /**
     * Signs each signalling message for verification by the peer.
     * Returns a base64-encoded string signature.
     */
    signMessage(message: string): Promise<string>;

    /**
     * Verifies a signalling message signature received by the peer.
     * The signature must be base64-encoded.
     */
    verifyMessage(base64Signature: string, message: string): Promise<boolean>;
}

File: peer-go/peer-config/peer_auth.go
package peerconfig

type PeerAuth interface {
	// Signs each signalling message for verification by the peer.
	// Returns a base64-encoded string signature.
	SignMessage(message string) (string, error)

	// Verifies a signalling message signature received by the peer.
	// The signature must be base64-encoded.
	VerifyMessage(base64Signature string, message string) bool

	// Generates a one-time cryptographically strong random string.
	GenerateNonce() string
}
